smac/intensification/intensification.py

```diff
@@ -188,6 +194,10 @@ def intensify(self, challengers, incumbent, run_history, aggregate_func,
 
                 # Line 10
                 missing_runs = list(inc_inst_seeds - chall_inst_seeds)
+                
+                if (not first_run) and missing_runs:
+                    first_run = True
+                    chall_indx += 1
```

Comment:
I think this should be further down where the algorithm is actually called. Right now it could happen that two challengers have already been run on all instances and thus not considered. In that cases we want to continue iterating the challengers until we find one that was not yet evaluated, rights?